lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1719,6 +1722,43 @@ def test_activate_block_id(self):
         )
         self.assertIn("Activate Block ID: test_block_id", response.content)
 
+    def test_negative_position(self):
+        self._assert_redirected_view_at_position(-5, 1)
+
+    def test_positive_position(self):
+        self._assert_redirected_view_at_position(2, 2)
+
+    def test_large_positive_position(self):
+        self._assert_redirected_view_at_position(8, 1)
+
+    def test_zero_position(self):
+        self._assert_redirected_view_at_position(0, 1)
+
+    def _assert_redirected_view_at_position(self, input_position, expected_position):
```

Comment:
Tests should have three phases:

1. Set-up the conditions (create the course, enroll the student)
2. Perform an action (call the view)
3. Check the result.

Putting everything in the assertion makes it harder to follow what the test is actually checking for.  Splitting this up so the three phases are reflected in the code will make it easier for later developers to see what the code is doing.  Put these tests into their own TestCase class, do the creation in a setUp(self) method, and split the _assert* method into a get_course_vertical method that takes the input position and returns a response, and an `_assert*` method that takes the response and the expected position.

Also rename the assert method.  You don't actually check for redirect.